src/main/java/cs321/btree/BTree.java

```diff
@@ -22,31 +26,56 @@
     private static final String NODE_FILE_EXTENSION = ".node";
 
     private int maxFilesPerDirectory = 1000;
-    private long nextGuid = 1; // Only use positives
-    private BTreeNode rootNode;
+    private long nextGuid = 0; // Only use positives
+    private long rootGuid;
     private String treeDirectory;
     private int degree;
-    private int size;
+    private long numKeys;
+    private int numNodes;
 
     public BTree(String treeDirectory, int degree) {
         // Will always create a new BTree with constructor
 
         try {
-            if (Files.exists(Paths.get(treeDirectory)))
-                deleteDirectoryRecursive(Paths.get(treeDirectory));
-            
+        	if(Paths.get(treeDirectory).toFile().exists()) {
+            	deleteDirectoryRecursive(Paths.get(treeDirectory));
+        	}
             Files.createDirectory(Paths.get(treeDirectory));
         } catch (Exception e) {
             e.printStackTrace();
         }
 
         this.treeDirectory = treeDirectory;
         this.degree = degree;
-        this.size = 0;
+        numKeys = 0;
 
-        rootNode = new BTreeNode(0, degree);
+        rootGuid = allocateNode();
+        if (rootGuid == -1) {
+            throw new RuntimeException();
+        }
+        
 
-        updateMetaFile();
```

Comment:
This still needs to be hear. And additionally any time the main BTree contents are changed (root node, size, etc.)